tests/test_splitter.py

```diff
@@ -255,31 +255,110 @@ def test_split_for_classification_gives_error(self):
                 splits = [("train_", 0.5), ("val", 0.2), ("test", 0.3)]
                 splitter.ClassificationSplit(source, splits)
 
-    def test_split_for_matching_reid(self):
+    def test_split_for_reidentification_with_label(self):
```

Comment:
Let's split this test into several smaller and independent ones. It is hard to track what's going on here.

Reply:
Okay. I'll split it.